src/circuit.rs

```diff
@@ -1550,7 +1560,9 @@ mod tests {
                 let (circuit, instance) =
                     generate_circuit_instance(is_native_asset, split_flag, &mut rng);
 
-                check_proof_of_orchard_circuit(&circuit, &instance, true);
+                let should_pass = !matches!((is_native_asset, split_flag), (true, true));
```

Comment:
very cool, let's emphasize the not:
`!(matches!((is_native_asset, split_flag), (true, true)));`

Reply:
Done